### PR TITLE
Mingshan/maxpool3d

### PR DESCRIPTION
**Pull request by mingshan-wang**
Add MaxPool3D support on the bridge.

Uses the helper function added by @sindhu-nervana 's PR. Will resolve conflicts once @sindhu-nervana 's Conv3D PR is merged.

### CODE REVIEW

File: test/test_nn_ops.cpp [hunk 1]
```diff
@@ -1079,6 +1116,62 @@ TEST(NNOps, L2Loss) {
...
+  vector<int> static_input_indexes = {};
+
+  for (auto const& input_size : input_sizes) {
+    Scope root = Scope::NewRootScope();
+
+    Tensor input_data(DT_FLOAT, TensorShape(input_size));
+    AssignInputValuesRandom<float>(input_data, -10, 10);
+
+    vector<int> filter = {1, 2, 2, 2, 1};
+    vector<int> stride = {1, 1, 1, 1, 1};
```
Comment (sayantan-nervana) [inline test/test_nn_ops.cpp]: filter size and stride can also be looped over for a couple of values

Comment (mingshan-wang) [reply test/test_nn_ops.cpp]: Yes, will add in more complicated test cases.

File: test/python/tensorflow/python_tests_list.txt [hunk 2]
```diff
@@ -112,6 +112,35 @@ conv_ops_test.Conv2DTest.testInputGradientValidPaddingStrideTwo
...
+conv_ops_3d_test.Conv3DTest.testInputGradientSamePaddingStrideThree 
+conv_ops_3d_test.Conv3DTest.testInputGradientSamePaddingStrideTwo 
+conv_ops_3d_test.Conv3DTest.testInputGradientValidPaddingStrideOne 
+conv_ops_3d_test.Conv3DTest.testInputGradientValidPaddingStrideThree
+conv_ops_3d_test.Conv3DTest.testInputGradientValidPaddingStrideTwo 
+conv_ops_3d_test.Conv3DTest.testKernelSizeMatchesInputSize 
+#conv_ops_3d_test.Conv3DTest.testKernelSmallerThanStride 
+conv_ops_3d_test.Conv3DTest.test_session 
+
 conv_ops_test.DepthwiseConv2DTest.testConv2D2x2Filter
```
Comment (sindhu-nervana) [inline test/python/tensorflow/python_tests_list.txt]: Could you also add the test from `conv3d_transpose_test` which tests Conv3D?

Comment (mingshan-wang) [reply test/python/tensorflow/python_tests_list.txt]: Sure, added now.

File: test/test_nn_ops.cpp (hunk 1 above)
Comment (sindhu-nervana) [reply test/test_nn_ops.cpp]: @mingshan-wang are you going to add more test cases in this PR or in a new one?

**Pull request merged**